application/src/test/java/org/thingsboard/server/controller/DeviceConnectivityControllerTest.java

```diff
@@ -309,15 +310,20 @@ public void testFetchGatewayLaunchCommands() throws Exception {
                 doGetTyped("/api/device-connectivity/gateway-launch/" + savedDevice.getId().getId(), new TypeReference<>() {
                 });
 
-        String expectedContainerName = deviceName.replaceAll("[^A-Za-z0-9_.-]", "");;
+        String expectedContainerName = deviceName.replace(" ", "_").replaceAll("[^A-Za-z0-9_.-]", "");
```

Comment:
I suggest to avoid using the device name at all. Using more than one gateway on the same host is unlikely. Experienced docker users will be able to resolve possible issues anyway.